### DISCUSSION TITLE
Vercel: No serverless pages were built

### DISCUSSION THREAD

**Discussion by caseypugh** [vercel/next.js#45875] [Help] [answered]
I recently upgraded from Nextjs 12 to `13.1.6` and updated `@sentry/nextjs@7.37.1` to the latest. I tested these changes on a Vercel preview server and everything worked great. 

When I transitioned to production, my builds started failing saying: 

```
  ...
info  - Creating an optimized production build...
Done in 177.02s
Found next.config.js
  ... 
(strangely, Vercel proceeds to output my entire next.config.js to the logs) 
  ...
Error: No serverless pages were built.
```

Whenever I disable Sentry entirely in my `next.config.js`, my builds start working again which leads me to think Sentry is the issue (but I have no idea why). 

Again, this works on a preview build but not production which leads me to think there is an issue with my ENVs, but as far as i can tell there are no major changes that I can see that would disrupt the build process. I tried running a `next build` locally using my production env vars and it built successfully. 

I'd file this as a proper bug but I cant seem to recreate the issue in a sandbox. Has anyone ever experienced this or know what I might be doing wrong?

**Comment by icyJoseph** [comment]
While going from 12 to 13 the next.js configuration API was standardized, and the plugins ought to conform to the expected API.

Look through the last answers here: https://github.com/getsentry/sentry-javascript/issues/6336#issuecomment-1332080106 and let us know how it goes.

**Comment by caseypugh** [comment, marked as answer]
@icyJoseph thanks so much! You lead me down a path of discovering that `withSentryConfig` sometimes returns a function. These changes fixed my `next.config.js` setup in production. 

```js
const nextConfig = { ... }
module.exports = (phase, defaultConfig) => {

  // ... other configurations redacted 

  let sentryExtendedConfig = withSentryConfig(nextConfig, {
    silent: true
  })

  // withSentryConfig can return either an config object or a function. Because we're
  // not sure which one it returns, we must check and handle each case.
  if (typeof sentryExtendedConfig === "function") {
    return sentryExtendedConfig(phase, defaultConfig);
  }
  else {
    return sentryExtendedConfig;
  }
}
```

Found via this issue https://github.com/getsentry/sentry-javascript/issues/6782#issuecomment-1398915528
